internal/provider/group_resource.go

```diff
@@ -77,17 +77,15 @@ func (r *GroupResource) Schema(ctx context.Context, req resource.SchemaRequest,
 				MarkdownDescription: "The unique name of the group.",
 				Required:            true,
 				Validators: []validator.String{
-					stringvalidator.LengthBetween(1, 36),
-					stringvalidator.RegexMatches(nameValidRegex, "Group names must be alpahnumeric with hyphens."),
+					codersdkvalidator.Name(),
```

Comment:
Group names need to use `codersdk.GroupNameValid` (which we recently bumped to like 255 characters)